src/KafkaFlow.Retry/Durable/Repository/RetryDurableQueueRepository.cs

```diff
@@ -257,6 +243,28 @@ private async Task<AddIfQueueExistsResult> AddIfQueueExistsAsync(IMessageContext
             }
         }
 
+        private RetryQueueItemMessage CreateRetryQueueItemMessage(IMessageContext context)
+        {
+            var partitionKey = default(byte[]);
```

Comment:
If we assume here the partitionKey as null (default(byte[])), we will get an SQL exception in the table `ItemMessage`, because the partition key field is not nullable. Furthermore, the kafka flow does not support produce a message with the null partition key as far as I know. PLease check the produce method.

Reply:
Maybe a `new byte[0]` solves the issue?

Let me also suggest to use a ternary condition here:

```csharp
var partitionKey = context.Message.Key is object ? Encoding.ASCII.GetBytes(JsonConvert.SerializeObject(context.Message.Key)) : new byte[0];
```

Because creating an empty byte array always incurs a memory cost. I know, almost negligible, but since we are here 🙂

Reply:
Yes, I agree with you @luispfgarces